src/views/components/package-file-browser.html

```diff
@@ -183,40 +203,92 @@
-										this.set('filesLoadingMessage', errorHtml);
-									});
-								}
-
-								this.set('filesLoadingMessage', errorHtml);
-							} else {
-								this.set('filesLoadingMessage', defaultError);
+					this.set('files', []);
+					this.set('filesLoadingMessage', '');
+					let packageType = this.get('type');
+					let packageName = this.get('name');
+
+					http.fetchPackageEntrypoints(packageType, packageName, version).then((data) => {
+						this.set('default', data.entrypoints.js ? data.entrypoints.js.file : '');
+					});
+
+					// Load package files.
+					http.fetchPackageFiles(packageType, packageName, version).then((data) => {
+						this.set('files', data.files);
+					}).catch((error) => {
+						let defaultError = 'Failed to load the file listing.';
+
+						if (error.responseJSON && error.responseJSON.message) {
+							let errorHtml = defaultError + '<br>' + error.responseJSON.message;
+
+							if (error.responseJSON.message.indexOf('size exceeded') !== -1 && this.get('github')) {
+								http.findProjectIssue(this.get('github.user'), this.get('github.project'), `${packageType === 'npm' ? 'npm package' : 'GitHub release'} too big for jsDelivr`).then((data) => {
+									errorHtml += `<br><br>Consider <a target="_blank" rel="noopener noreferrer" href="${data.items.length ? data.items[0].html_url : this.get('issueTooBig')}">asking the author to remove unnecessary files</a> to make the package smaller.`;
+								}).always(() => {
+									errorHtml += `<strong><br><br>You may still find the files you need in the <a target="_blank" rel="noopener noreferrer" href="https://github.com/${this.get('github.user')}/${this.get('github.project')}/">GitHub repository</a> and include them in your project like this:<br><code>https://cdn.jsdelivr.net/gh/${this.get('github.user')}/${this.get('github.project')}@version/file</code></strong><br><br>Read more about <a target="_blank" href="/features#gh">GitHub CDN</a>, which can load individual files even if the overall package size exceeds the limit.`;
+									this.set('filesLoadingMessage', errorHtml);
+								});
 							}
+
+							this.set('filesLoadingMessage', errorHtml);
+						} else {
+							this.set('filesLoadingMessage', defaultError);
+						}
+					});
+
+					// fetch package File stats to get downloads amount per file
+					http.fetchPackageFileStats(packageType, packageName, version).then((response) => {
+						let downloadsPerFile = response.reduce((downloadsPerFile, fileStats) => {
+							let fileName = fileStats.name.substr(fileStats.name.lastIndexOf('/') + 1);
```

Comment:
This logic here won't work with files that have the same name but are in different directories. In the file browser you have a `path` variable which you'll need to use too to access these stats.

Reply:
You are right. Fixed.